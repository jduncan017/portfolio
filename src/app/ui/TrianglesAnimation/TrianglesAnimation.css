.triangles-outer-container {
	height: 100%;
	left: 0;
	overflow: hidden;
	position: absolute;
	top: 0;
	width: 100%;
}

.triangle-container {
	height: 100%;
	left: 50%;
	position: relative;
	top: 50%;
	transform: translate(-50%, -50%);
	width: 100%;
}

.triangle {
	animation: triangle-animation 15s linear infinite;
	fill: none;
	stroke: #FFF;
	stroke-width: 13;
	transform-origin: center;
}

.triangle-1 {
	animation-delay: 0s;
}

.triangle-2 {
	animation-delay: -1s;
}

.triangle-3 {
	animation-delay: -2s;
}

.triangle-4 {
	animation-delay: -3s;
}

.triangle-5 {
	animation-delay: -4s;
}

.triangle-6 {
	animation-delay: -5s;
}

.triangle-7 {
	animation-delay: -6s;
}

/* ************************************* */

/* .triangle-1 {
	animation-delay: 0s;
}

.triangle-2 {
	animation-delay: -0.5s;
}

.triangle-3 {
	animation-delay: -1s;
}

.triangle-4 {
	animation-delay: -1.5s;
}

.triangle-5 {
	animation-delay: -2s;
}

.triangle-6 {
	animation-delay: -2.5s;
}

.triangle-7 {
	animation-delay: -3s;
}

.triangle-8 {
	animation-delay: -3.5s;
}

.triangle-9 {
	animation-delay: -4s;
}

.triangle-10 {
	animation-delay: -4.5s;
}

.triangle-11 {
	animation-delay: -5s;
}

.triangle-12 {
	animation-delay: -5.5s;
}

.triangle-13 {
	animation-delay: -6s;
}

.triangle-14 {
	animation-delay: -6.5s;
} */

@keyframes triangle-animation {

	0% {
		opacity: .3;
		transform: scale(0) rotate(0deg);
	}

	100% {
		opacity: 0;
		transform: scale(3) rotate(45deg);
	}

}
