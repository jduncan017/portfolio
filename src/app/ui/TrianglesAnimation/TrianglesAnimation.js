import React from 'react';
import './TrianglesAnimation.css';

export const TrianglesAnimation = () => {

  const buildTringles = () => {
    let triangles = []
    for ( let i = 1; i <= 7; i++ ) {
      triangles.push(<polygon className={`triangle triangle-${i}`} key={i} points='500,200 759,650 241,650' />)
    }
    return (
      <svg className='triangle-container' viewBox='0 0 1000 1000'>
        { triangles }
      </svg>
    )
  }

  return (
    <section className='triangles-outer-container'>
      { buildTringles() }
    </section>
  );
}
