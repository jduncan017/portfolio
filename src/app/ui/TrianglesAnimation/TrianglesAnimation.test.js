import React from 'react';
import { TrianglesAnimation } from './TrianglesAnimation';
import { shallow } from 'enzyme';

describe('TrianglesAnimation', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TrianglesAnimation />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
