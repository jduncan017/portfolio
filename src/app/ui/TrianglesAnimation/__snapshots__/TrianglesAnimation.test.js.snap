// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrianglesAnimation should match the snapshot 1`] = `
<section
  className="triangles-outer-container"
>
  <svg
    className="triangle-container"
    viewBox="0 0 1000 1000"
  >
    <polygon
      className="triangle triangle-1"
      key="1"
      points="500,200 759,650 241,650"
    />
    <polygon
      className="triangle triangle-2"
      key="2"
      points="500,200 759,650 241,650"
    />
    <polygon
      className="triangle triangle-3"
      key="3"
      points="500,200 759,650 241,650"
    />
    <polygon
      className="triangle triangle-4"
      key="4"
      points="500,200 759,650 241,650"
    />
    <polygon
      className="triangle triangle-5"
      key="5"
      points="500,200 759,650 241,650"
    />
    <polygon
      className="triangle triangle-6"
      key="6"
      points="500,200 759,650 241,650"
    />
    <polygon
      className="triangle triangle-7"
      key="7"
      points="500,200 759,650 241,650"
    />
  </svg>
</section>
`;
